### PR TITLE
perf: remove `parse-body` from the server runtime

### PR DESCRIPTION
**Pull request by feedthejim**
This PR removes the `raw-body` dependency from the server runtime. It still is used in the pages runtime but at least if you're only using app, you won't bundle your server function with it.

### CODE REVIEW

**Comment by feedthejim** [comment]
Current dependencies on/for this PR:
* canary
  * **PR #57280** <a href="https://app.graphite.dev/github/pr/vercel/next.js/57280" target="_blank">Graphite</a>  👈
    * **PR #57281** <a href="https://app.graphite.dev/github/pr/vercel/next.js/57281" target="_blank">Graphite</a> 
      * **PR #57282** <a href="https://app.graphite.dev/github/pr/vercel/next.js/57282" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/vercel/next.js/57280?utm_source=stack-comment).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 10-23-deprecate_parseBody | Change |
| - | - | - | - |
| buildDuration | 10.8s | 10.7s | N/A |
| buildDurationCached | 6.2s | 6.2s | N/A |
| nodeModulesSize | 174 MB | 174 MB | N/A |
| nextStartRea..uration (ms) | 516ms | 519ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 10-23-deprecate_parseBody | Change |
| - | - | - | - |
| 199-HASH.js gzip | 27.9 kB | 27.9 kB | N/A |
| 3f784ff6-HASH.js gzip | 53.2 kB | 53.2 kB | N/A |
| 99.HASH.js gzip | 182 B | 182 B | ✓ |
| framework-HASH.js gzip | 45.5 kB | 45.5 kB | ✓ |
| main-app-HASH.js gzip | 254 B | 252 B | N/A |
| main-HASH.js gzip | 33 kB | 33 kB | N/A |
| webpack-HASH.js gzip | 1.75 kB | 1.75 kB | N/A |
| Overall change | 45.7 kB | 45.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 10-23-deprecate_parseBody | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 10-23-deprecate_parseBody | Change |
| - | - | - | - |
| _app-HASH.js gzip | 206 B | 205 B | N/A |
| _error-HASH.js gzip | 182 B | 180 B | N/A |
| amp-HASH.js gzip | 506 B | 505 B | N/A |
| css-HASH.js gzip | 322 B | 323 B | N/A |
| dynamic-HASH.js gzip | 2.59 kB | 2.59 kB | ✓ |
| edge-ssr-HASH.js gzip | 260 B | 259 B | N/A |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 369 B | 369 B | ✓ |
| image-HASH.js gzip | 4.38 kB | 4.38 kB | N/A |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.67 kB | 2.67 kB | N/A |
| routerDirect..HASH.js gzip | 316 B | 318 B | N/A |
| script-HASH.js gzip | 385 B | 384 B | N/A |
| withRouter-HASH.js gzip | 319 B | 319 B | ✓ |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.99 kB | 3.99 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 10-23-deprecate_parseBody | Change |
| - | - | - | - |
| _buildManifest.js gzip | 484 B | 482 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 10-23-deprecate_parseBody | Change |
| - | - | - | - |
| index.html gzip | 528 B | 529 B | N/A |
| link.html gzip | 541 B | 541 B | ✓ |
| withRouter.html gzip | 523 B | 522 B | N/A |
| Overall change | 541 B | 541 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 10-23-deprecate_parseBody | Change |
| - | - | - | - |
| edge-ssr.js gzip | 95.7 kB | 95.7 kB | N/A |
| page.js gzip | 158 kB | 158 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 10-23-deprecate_parseBody | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 625 B | 624 B | N/A |
| middleware-r..fest.js gzip | 150 B | 151 B | N/A |
| middleware.js gzip | 22.9 kB | 22.9 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 1.92 kB | 1.92 kB | ✓ |

</details>

</details>
Commit: 47baef7030e281d9298b11b0a6ac545235e13e75

**Review by huozhi — DISMISSED** [review]

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 9092ad4cfbeb42bcd21291825f8b183df7a59418

`pnpm test test/integration/edge-runtime-configurable-guards/test/index.test.js` 
- Edge runtime configurable guards > Multiple functions with different configurations > warns in dev for allowed code

<details>
<summary>Expand output</summary>

  ● Edge runtime configurable guards › Multiple functions with different configurations › warns in dev for allowed code

    expect(received).toContain(expected) // indexOf

    Expected substring: "Dynamic Code Evaluation (e. g. 'eval', 'new Function') not allowed in Edge Runtime"
    Received string:    "   ▲ Next.js 13.5.7-canary.17
       - Local:        http://localhost:45763·
     ⨯ Middleware cannot be used with \"output: export\". See more info here: https://nextjs.org/docs/advanced-features/static-html-export
     ⨯ API Routes cannot be used with \"output: export\". See more info here: https://nextjs.org/docs/advanced-features/static-html-export
     ✓ Ready in 2.3s
     ○ Compiling / ...
     ✓ Compiled / in 788ms (189 modules)
    "

      84 |       await waitFor(500)
      85 |       expect(res.status).toBe(200)
    > 86 |       expect(context.logs.output).toContain(
         |                                   ^
      87 |         `Dynamic Code Evaluation (e. g. 'eval', 'new Function') not allowed in Edge Runtime`
      88 |       )
      89 |     })

      at Object.toContain (integration/edge-runtime-configurable-guards/test/index.test.js:86:35)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: packages/next/src/server/web/sandbox/context.ts [hunk 1]
```diff
@@ -5,10 +5,10 @@ import type {
...
 import type { UnwrapPromise } from '../../../lib/coalesced-function'
 import { AsyncLocalStorage } from 'async_hooks'
-import {
-  decorateServerError,
-  getServerError,
-} from 'next/dist/compiled/@next/react-dev-overlay/dist/middleware'
+// import {
+//   decorateServerError,
+//   getServerError,
+// } from 'next/dist/compiled/@next/react-dev-overlay/dist/middleware'
```
Comment (ijjk) [inline packages/next/src/server/web/sandbox/context.ts:11]: Why leave these?

**Pull request merged**